### PR TITLE
Small improvements to nlsys, bdalg

### PR DESCRIPTION
**Pull request by murrayrm**
This PR provides some small changes to nonlinear I/O systems process:

* Allows system name to be overridden in `linearize`, even if `copy_names` is `False`.
* Allows renaming of system/signal names in bdalg functions:
  ```
  sys = ct.series(sys1, sys2, inputs='u', outputs='y')
  ```
* New `update_names` method for I/O systems that allows signal and system names to be updated.
* Refactoring of code for processing `x0`, `u0` keywords in `linearize` and `input_output_response` to provide common functionality in allowing concatenation of lists and zero padding ("vector element processing").
* Improved error messages when `x0` and `u0` don't match the expected size.
* If no output function is given in `nlsys`, which provides full state output, the output signal names are set to match the state names.
* Updated unit tests and documentation (including a new section on "vector element processing").

### CODE REVIEW

File: control/nlsys.py [hunk 1]
```diff
@@ -2572,18 +2547,57 @@ def interconnect(
...
+
+    """
+    # Allow and expand list
+    if isinstance(arg, (tuple, list)):
+        val_list = []
+        for i, v in enumerate(arg):
+            v = np.array(v).reshape(-1)             # convert to 1D array
+            val_list += v.tolist()                  # add elements to list
+        val = np.array(val_list)
+    elif np.isscalar(arg) and size is not None:     # extend scalars
```
Comment (roryyorke) [inline control/nlsys.py:2586]: does this funciton need to handle the case when np.isscalar(arg) and size *is* None, e.g., do `val = np.array([arg])` ?

**Review by roryyorke — COMMENTED** [review]
Functionality looks good.  Minor suggestions and remarks.

File: control/nlsys.py (hunk 1 above)
Comment (murrayrm) [reply control/nlsys.py:2586]: Added this case.

File: control/nlsys.py [hunk 2]
```diff
@@ -1467,7 +1458,16 @@ def input_output_response(
         # Use the input time points as the output time points
         t_eval = T
 
-    # If we were passed a list of input, concatenate them (w/ broadcast)
+    #
+    # Process input argument
+    #
+    # The input argument is interpreted very flexibly, allowing the
+    # use of listsa and/or tuples of mixed scalar and vector elements.
```
Comment (slivingston) [inline control/nlsys.py:1465]:
```suggestion
    # use of lists and/or tuples of mixed scalar and vector elements.
```

File: doc/iosys.rst [hunk 3]
```diff
@@ -191,7 +191,86 @@ Additional features
...
+
+* Lists of values are concatenated into a single vector.  This is
+  often used when you have an interconnected system and you need to
+  specify the initial condition or input value for each subsystem
+  (e.g., [X1eq, X2eq, ...]).
+
+* Vector elements are zero padded to the required length. If you
+  specify only a portion of the values for states or inputs, the
+  remaining values are taken as zero.  (If the final element in the
+  given vector is non-zero, a warning is issues.)
```
Comment (slivingston) [inline doc/iosys.rst:219]:
```suggestion
  given vector is non-zero, a warning is issued.)
```

File: doc/iosys.rst [hunk 4]
```diff
@@ -191,7 +191,86 @@ Additional features
...
+To linearize the system around the origin, we can utilize the scalar
+processing feature of vector elements::
+
+  P = proc.linearize(0, 0)
+
+In this command, the states and the inputs are broadcast to the size of the
+state and input vectors, respectively.
+
+If we want to linearize the closed loop system around a process state
+``x0`` (with two elemenst) and an estimator state ``0`` (for both states),
```
Comment (slivingston) [inline doc/iosys.rst:254]:
```suggestion
``x0`` (with two elements) and an estimator state ``0`` (for both states),
```

File: doc/iosys.rst [hunk 5]
```diff
@@ -191,7 +191,86 @@ Additional features
...
+we can use the list processing feature::
+
+  H = clsys.liniearize([x0, 0], 0)
```
Comment (slivingston) [inline doc/iosys.rst:257]:
```suggestion
  H = clsys.linearize([x0, 0], 0)
```

File: doc/iosys.rst [hunk 6]
```diff
@@ -191,7 +191,86 @@ Additional features
...
+
+Note that this also utilizes the zero-padding functionality, since the
+second argument in the list ``[x0, 0]`` is a scalar and so the vector
+``[x0, 0]`` only has three elements instead of the required four.
+
+To run an input/output simulation with a sinsoidal signal for the first
```
Comment (slivingston) [inline doc/iosys.rst:263]:
```suggestion
To run an input/output simulation with a sinusoidal signal for the first
```

**Pull request merged**